File: backend/internal/constants/inspection.go
package constants

// 巡检周期
const (
	CycleDaily     = "daily"
	CycleWeekly    = "weekly"
	CycleMonthly   = "monthly"
	CycleQuarterly = "quarterly"
	CycleYearly    = "yearly"
)

// ValidCycles 合法巡检周期
var ValidCycles = map[string]bool{
	CycleDaily: true, CycleWeekly: true, CycleMonthly: true,
	CycleQuarterly: true, CycleYearly: true,
}

// 设施状态
const (
	FacilityStatusAvailable = "available"
	FacilityStatusDisabled  = "disabled"
)

// ValidFacilityStatuses 合法设施状态
var ValidFacilityStatuses = map[string]bool{
	FacilityStatusAvailable: true, FacilityStatusDisabled: true,
}

// 巡检任务种类
const (
	TaskKindRoutine = "routine"
	TaskKindRecheck = "recheck"
)

// 巡检任务状态
const (
	TaskStatusPending       = "pending"
	TaskStatusClaimed       = "claimed"
	TaskStatusDone          = "done"           // 常规巡检无隐患（终态）
	TaskStatusHazard        = "hazard"         // 巡检发现隐患，设施停用、已建工单（终态）
	TaskStatusRecheckFailed = "recheck_failed" // 复检未通过，保持停用、已续建工单（终态）
	TaskStatusRestored      = "restored"       // 复检通过，设施恢复可用（终态）
)

// ValidTaskStatuses 合法任务状态
var ValidTaskStatuses = map[string]bool{
	TaskStatusPending: true, TaskStatusClaimed: true, TaskStatusDone: true,
	TaskStatusHazard: true, TaskStatusRecheckFailed: true, TaskStatusRestored: true,
}

// TerminalTaskStatuses 终态任务，结果不允许被后续调整改写
var TerminalTaskStatuses = map[string]bool{
	TaskStatusDone: true, TaskStatusHazard: true,
	TaskStatusRecheckFailed: true, TaskStatusRestored: true,
}

// 巡检结果
const (
	ResultNormal = "normal" // 常规巡检正常
	ResultHazard = "hazard" // 常规巡检发现隐患
	ResultPass   = "pass"   // 复检通过
	ResultFail   = "fail"   // 复检未通过
)

File: frontend/src/constants/inspection.ts
import type {FacilityStatus, InspectionCycle, InspectionTaskKind, InspectionTaskStatus} from '../types';

export const FACILITY_STATUS: Record<Uppercase<FacilityStatus>, FacilityStatus> = {
  AVAILABLE: 'available', DISABLED: 'disabled'
};
export const facilityStatusText: Record<FacilityStatus, string> = {
  available: '可用', disabled: '停用'
};

export const CYCLE_TEXT: Record<InspectionCycle, string> = {
  daily: '每日', weekly: '每周', monthly: '每月', quarterly: '每季度', yearly: '每年'
};

export const TASK_KIND_TEXT: Record<InspectionTaskKind, string> = {
  routine: '常规巡检', recheck: '维修复检'
};

export const INSPECTION_TASK_STATUS: Record<Uppercase<InspectionTaskStatus>, InspectionTaskStatus> = {
  PENDING: 'pending', CLAIMED: 'claimed', DONE: 'done',
  HAZARD: 'hazard', RECHECK_FAILED: 'recheck_failed', RESTORED: 'restored'
};
export const inspectionTaskStatusText: Record<InspectionTaskStatus, string> = {
  pending: '待巡检',
  claimed: '已接单',
  done: '巡检正常',
  hazard: '发现隐患·停用',
  recheck_failed: '复检未过·停用',
  restored: '复检通过·恢复'
};
// 任务状态 -> Element Plus 标签颜色（与 RepairStatusBadge 风格一致）。
export const inspectionTaskStatusKind: Record<InspectionTaskStatus, 'warning'|'primary'|'success'|'danger'|'info'> = {
  pending: 'warning',
  claimed: 'primary',
  done: 'success',
  hazard: 'danger',
  recheck_failed: 'danger',
  restored: 'success'
};
export const facilityStatusKind: Record<FacilityStatus, 'success'|'danger'> = {
  available: 'success', disabled: 'danger'
};
